### ISSUE TITLE
Listening to music while in Airplane Mode DELETES MY LOGIN DATA

### ISSUE THREAD

**Issue by deejf** [tgwizard/sls#290]
Just upgraded to 1.5.0 through F-Droid, and this morning checked my scrobble list while I was playing music. Most of the time, my phone is in Airplane Mode so that I only get bothered by calls when I choose to be. 

Turns out, with the upgrade, this means that I get an obnoxious warning that "credentials REQUIRED to work properly", *and* I now find that, apparently because the credential verification didn't happen at the very instant the program demanded, my usernames and passwords for both services were **deleted**. Just gone.

Do I even need to explain how rude and annoying this is? I'm going to have to manually enter two logins and two passwords every freaking day now? Seriously?

There *used* to be a preference for "scrobble only over wifi", but that appears to be gone, or else it is hidden and was changed with the upgrade without asking me if I wanted to change it. Which, obviously, I didn't.

**UPDATE:** OK, I found the "only over wifi" option and set it to that. Again. It's still annoying that my prefs didn't carry through the upgrade, and that my credentials were "helpfully" deleted for me.

**Comment by HumbleBeeBumbleBee** [comment]
This is the new api 2.0. It is supposed to be better than the previous. Apologies for this but it would be necessary to relogin to use the new login system. Otherwise when the older api is shutdown  your Scrobbles would just sit in the cache and if nobody is there to update the app everyone would be upset.

**Comment by deejf** [comment]
Having to relogin once, okay, I can understand that.

Having to re-enter *all* of my login information every day? That is not good. Deleting information already entered just because it can't find a server at one moment is a *bad* thing.

**Comment by HumbleBeeBumbleBee** [comment]
So your device is requiring religion every day?

**Comment by HumbleBeeBumbleBee** [comment]
Can I ask what type of phone and android version?

**Comment by HumbleBeeBumbleBee** [comment]
SLS is built to only reset login information if the session key is invalid.

**Comment by deejf** [comment]
Was I unclear? Let me try again.

My phone was in Airplane Mode. That is, it was not connected to any network, neither phone nor data.

When SLS noticed no network was available, it deleted my "credentials", stopped recording scrobbles, and told me credentials were required.

This apparently means that the session key is "invalid" when no network is available. 

Since I leave my phone in Airplane Mode a lot, that would be a Bad Thing for me.

**Comment by HumbleBeeBumbleBee** [comment]
This happens every time?

**Comment by deejf** [comment]
Well, since I found the "only update over wifi" preference again, hopefully I won't have to find out. I might even activate the manual-scrobbling-only preference, just to be sure.

**Comment by HumbleBeeBumbleBee** [comment]
https://play.google.com/apps/testing/com.adam.aslfms

**Comment by HumbleBeeBumbleBee** [comment]
@deejf  Are you still available to help with this issue?
I was hoping you could test it for me.

**Comment by HumbleBeeBumbleBee** [comment]
Do you know how to use adb?
